include/turtle/detail/mock_impl.hpp

```diff
@@ -0,0 +1,108 @@
+// http://turtle.sourceforge.net
+//
+// Copyright Mathieu Champlon 2008
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef MOCK_MOCK_IMPL_HPP_INCLUDED
+#define MOCK_MOCK_IMPL_HPP_INCLUDED
+
+#include "function.hpp"
+#include "functor.hpp"
+#include "parameter.hpp"
+#include "signature.hpp"
+#include "type_name.hpp"
+#include <boost/preprocessor/repetition/repeat.hpp>
+#include <boost/preprocessor/stringize.hpp>
+
+// Internal compatibility macro if function signature is passed via BOOST_IDENTITY_TYPE
+// TODO: Remove support for doing that and move remove_pointer_t to MOCK_PROTECT_FUNCTION_SIG
+#define MOCK_FUNCTION_TYPE(...) std::remove_pointer_t<__VA_ARGS__>
+
+#define MOCK_HELPER(t) t##_mock(mock::detail::root, BOOST_PP_STRINGIZE(t))
```

Comment:
it might be too big of an "ask" for this change, but would you consider adding some brief documentation to the macro parameters?

Even giving them names that aren't 1 letter would be a big enhancement to understandability.

Reply:
Will probably do as a drive-by when I touch those. TBH I don't really took the time to understand each and every macro ;-)
And as this is implementation detail, documentation is optional 🤷 
Jokes aside: Might do later, this is just refactoring and making sure the public interface is documented.